andela/ah-backend-sparta
# #164798220 Highlight and Comment on texts in Articles

**What does this PR do?**

- Enables users to highlight and comment on an article

**Description of Task to be completed?**

**How should this be manually tested?**

- Fetch this branch git fetch origin `ft-highlight-comment-article-164798220`
- Checkout to the branch git checkout `ft-highlight-comment-article-164798220`
- Run `python manage.py makemigrations`
- Run `python manage.py migrate`
- Run `python manage.py runserver`
- Login a user POST /api/users/login/
- Create an article POST /api/articles/
- Create a comment POST /api/articles/<str:slug>/comments/
the format of your comment should include the area of the article to highlight as below:

```{
  "comment": {
    "body": "i want to comment on this article",
    "start_position":6,
     "end_position":12
  }
}
```

 **What are the relevant pivotal tracker stories?**

[#164798220](https://www.pivotaltracker.com/n/projects/2320796/stories/164798220)

**Screenshots**

<img width="1133" alt="Screenshot 2019-04-17 at 16 52 27" src="https://user-images.githubusercontent.com/40162086/56293660-32336980-6132-11e9-8d56-dfd501187754.png">
The start_position and end position show the area of the article that has been highlighted
